### ISSUE TITLE
Database migration failure after updating to 1.8.0 on mariadb

### ISSUE THREAD

**Issue by ruifung** [nextcloud/mail#4443]
### Expected behavior
Tell us what should happen
Migration should succeed

### Actual behavior
Tell us what happens instead
Migration fails as follows:
`Updating database schema

Updated database

Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'CREATE TABLE oc_mail_trusted_senders (id INT AUTO_INCREMENT NOT NULL, email VARCHAR(255) NOT NULL, user_id VARCHAR(64) NOT NULL, UNIQUE INDEX mail_trusted_sender_uniq (email, user_id), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_bin` ENGINE = InnoDB ROW_FORMAT = compressed': SQLSTATE[HY000]: General error: 1005 Can't create table `nextcloud`.`oc_mail_trusted_senders` (errno: 140 "Wrong create options")`

Attempting the same query with show warnings gets the following issue:
`InnoDB: PAGE_COMPRESSED table can't have ROW_TYPE=COMPRESSED`

In addition, the mariaDB knowledgebase recommends using page compression instead of compressed row type.
https://mariadb.com/kb/en/innodb-compressed-row-format/

### Mail app

**Mail app version:** 1.8.0

**Mailserver or service:** IMAP

### Server configuration

**Operating system**: Ubuntu 20.04 Raspberry Pi

**Web server:** Caddy

**Database:** 10.5.8-MariaDB-1:10.5.8+maria~focal-log

**PHP version:** 7.4 FPM

**Nextcloud Version:** 20.0.5

#### Client configuration
**Browser:** N/A

**Operating system:** N/A

**Comment by Ornias1993** [comment]
This happens again, again and again.
Needs to be solved one time and for all time.

**Comment by ruifung** [comment]
Actually, now that I take a look at this again, isn't this actually caused by nextcloud creating tables with a specified row format by  default?

https://github.com/nextcloud/server/blob/905e1918d2796b9a79025283cd6edf2c40f49d77/lib/private/DB/ConnectionFactory.php#L228

**Comment by Forza-tng** [comment]
> Actually, now that I take a look at this again, isn't this actually caused by nextcloud creating tables with a specified row format by default?
> 
> https://github.com/nextcloud/server/blob/905e1918d2796b9a79025283cd6edf2c40f49d77/lib/private/DB/ConnectionFactory.php#L228

Yes this is incompatible with using page compression. Page compression is, however, better in most ways, including less memory usage.

**Comment by ChristophWurst** [comment]
Let's continue this in https://github.com/nextcloud/server/issues/25436 as the Mail app is only *using* the migration system but has little to no influence on *how* tables are created.

**Comment by rsmithlal** [comment]
I'm getting this same issue too with my postgres db. All of a sudden I couldn't update mail because it says that this table isn't present in the schema. I can't find anywhere online where the DB migration that creates this table exists either in the mail app or nextcloud itself. Where is this coming from? I had to remove the mail app from the cli because it was forcing my nextcloud into maintenance mode and wasn't able to be updated because of the error. Can't reinstall it now.

**Comment by ChristophWurst** [comment]
https://github.com/nextcloud/mail/search?q=mail_trusted_senders+path%3Alib%2FMigration for the migrations. Are both migrations/versions present in your oc_migrations?
